src/entity.ts

```diff
@@ -1249,34 +1238,12 @@ export namespace entity {
       queryProto.startCursor = query.startVal;
     }
 
-    if (query.filters.length > 0) {
-      const filters = query.filters.map(filter => {
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        let value: any = {};
-
-        if (filter.name === '__key__') {
-          value.keyValue = entity.keyToKeyProto(filter.val);
-        } else {
-          value = entity.encodeValue(filter.val, filter.name);
-        }
-
-        return {
-          propertyFilter: {
-            property: {
-              name: filter.name,
-            },
-            op: OP_TO_OPERATOR[filter.op],
-            value,
-          },
-        };
-      });
-
-      queryProto.filter = {
-        compositeFilter: {
-          filters,
-          op: 'AND',
-        },
-      };
+    if (query.filters.length > 0 || query.newFilters.length > 0) {
+      const filters = query.filters.map(
+        filter => new PropertyFilter(filter.name, filter.op, filter.val)
+      );
+      const newFilters = query.newFilters;
+      queryProto.filter = AND(newFilters.concat(filters)).toProto();
```

Comment:
a little confused by this, what is this logic doing? Can you add some comments?

Reply:
Added comments to the code here.

This is mainly just a refactor. Old code is being moved into the `Filter` classes, but this new code does the same thing as the removed code. The functionality had to be pulled into the `Filter` classes to avoid creating duplicate code.